Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -1027,6 +1027,293 @@ function Remove-GroupFromCloudAdmins {
     Write-Output "Current Cloud Admin Members: $CloudAdminMembers"
 }
 
+<#
+    .Synopsis
+     Add a group from the external identity to the SRMAdmins group
+
+    .Parameter GroupName
+     The group in the customer external identity source to be added to SRMAdmins. Users in this group will have SRMAdmin access. Group name should be formatted without the domain name, e.g. group-to-give-access
+
+    .Parameter Domain
+     Name of the external domain that GroupName is in. If not provided, will attempt to locate the group in all the configured active directories. For example, MyActiveDirectory.Com
+
+    .Example
+    # Add the group named vsphere-admins to SRMAdmins
+     Add-GroupToSRMAdmins -GroupName 'srmgrp-admins'
+#>
+function Add-GroupToSRMAdmins {
+    [CmdletBinding(PositionalBinding = $false)]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    Param
+    (
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Name of the group to add to SRMAdmin')]
+        [ValidateNotNull()]
+        [string]
+        $GroupName,
+
+        [Parameter(Mandatory = $false)]
+        [string]
+        $Domain
+    )
+
+    $ExternalSources
+    $GroupToAdd
+    $Domain
+
+    try {
+        $ExternalSources = Get-IdentitySource -External -ErrorAction Stop
+    }
+    catch {
+        Write-Error $PSItem.Exception.Message -ErrorAction Continue
+        Write-Error "Unable to get external identity source" -ErrorAction Stop
```

Comment:
Use `throw` instead, formatting the message with the original error from the excetion.